health-checker/src/main/java/io/stargate/health/CPUGaugeMetricSet.java

```diff
@@ -0,0 +1,97 @@
+package io.stargate.health;
+
+import com.codahale.metrics.Gauge;
+import com.codahale.metrics.Metric;
+import com.codahale.metrics.MetricSet;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import oshi.SystemInfo;
+import oshi.hardware.CentralProcessor;
+import oshi.hardware.CentralProcessor.TickType;
+
+public class CPUGaugeMetricSet implements MetricSet {
+
+  final CentralProcessor processor;
+  private long tickTime;
+  private final long[] prevTicks;
+  private final long[] curTicks;
+
+  public CPUGaugeMetricSet() {
+    processor = new SystemInfo().getHardware().getProcessor();
+    this.prevTicks = new long[TickType.values().length];
+    this.curTicks = new long[TickType.values().length];
+  }
+
+  @Override
+  public Map<String, Metric> getMetrics() {
+    final Map<String, Metric> gauges = new HashMap<>();
+
+    gauges.put("load", (Gauge<Double>) this::getSystemCpuLoad);
+    gauges.put("user", (Gauge<Double>) () -> getTick("user"));
+    gauges.put("nice", (Gauge<Double>) () -> getTick("nice"));
+    gauges.put("sys", (Gauge<Double>) () -> getTick("sys"));
+    gauges.put("idle", (Gauge<Double>) () -> getTick("idle"));
+    gauges.put("iowait", (Gauge<Double>) () -> getTick("iowait"));
+
+    return Collections.unmodifiableMap(gauges);
+  }
+
+  private Double getSystemCpuLoad() {
+    long now = System.currentTimeMillis();
+    if (now - this.tickTime > 950) {
+      // Enough time has elapsed.
+      updateSystemTicks();
+    }
+
+    return processor.getSystemCpuLoadBetweenTicks(prevTicks);
+  }
+
+  private Double getTick(String tickType) {
+    long now = System.currentTimeMillis();
+    if (now - this.tickTime > 950) {
+      // Enough time has elapsed.
+      updateSystemTicks();
+    }
```

Comment:
Time threshold can be overrun in-between `getTick` calls for different metrics, thus their values will be based on different tick samples... not a huge deal, but might lead to oddities like percentages not adding up to `100`, I guess.

Reply:
Honestly for alerting and such I would want to deal with thresholds and trends. Answering things like "is there a spike in CPU usage when X event happens" or "are we using more than 80% of the CPU". Also, I'm on the fence of whether I want to expose all of these or not (like "nice" for example).

Reply:
For most use cases I'd imagine cpuload would be enough but I can see where iowait would be useful. There's actually more ticks available that I'm omitting
https://github.com/goodyinxin/eladmin/blob/4adea4ebe2c4ca6f1c691f64036cc96a4a10737b/eladmin-monitor/src/test/java/me/zhengjie/SystemInfoTest.java#L194-L202

Reply:
From that link it looks like we're missing some data in the calculation of `totalCPU` :thinking:

Reply:
Yeah I left those out since they're not being exposed. Was thinking we'd just calculate percent of CPU based on what's being exposed. I can put them back if you'd prefer.

Reply:
I believe the percentage calculation can be incorrect if we leave out those other numbers. I assume each CPU tick goes into one of those buckets, so we have to account for all of them in the total.

Reply:
Added all of them